src/services/textChanges.ts

```diff
@@ -451,33 +466,45 @@ namespace ts.textChanges {
                 return "";
             }
             const options = change.options || {};
-            const nonFormattedText = getNonformattedText(change.node, sourceFile, this.newLine);
+            let text: string;
+            const pos = change.range.pos;
+            const posStartsLine = getLineStartPositionForPosition(pos, sourceFile) === pos;
+            if (isArray(change.node)) {
+                const parts = change.node.map(n => this.getFormattedTextOfNode(n, sourceFile, pos, change.useIndentationFromFile, options));
+                text = parts.join(change.options.nodesSeparator);

```

Comment:
The default is `,` which is probably not sensible. Reworking this so that the external surface area requires a separator if multiple new nodes are provided